pycoal/mineral.py

```diff
@@ -362,3 +365,96 @@ def convert(cls, data_dir="", db_file="", hdr_file=""):
         library = aster_database.create_envi_spectral_library(spectrum_ids, band_info)
 
         library.save(hdr_file)
+
+class SpectralToAsterConversion:
+    
+    def __init__(self):
+        """
+            This class provides a method for converting Spectral .txt files into
+            ASTER Library Version 2.0 .txt files
+            
+            Args:
+                none
+            """
+        pass
+    
+    @classmethod
+    def convert(cls, library_filename=""):
+        """
+            This class method converts a Spectral version 7 .txt file into
+            an ASTER Library Version 2.0 .spectrum.txt file
+            
+            Spectral Library Version 7 can be downloaded here https://speclab.cr.usgs.gov/spectral-lib.html
+            
+            How to use on entire USGS Spectral Library Version 7:
+            1. Move all .txt files in Spectral Library Version 7 to one directory named 'usgs_splib07_modified'
+            2. Place this directory('usgs_splib07_modified') you just made in the pycoal/examples directory
+            3. run splib07_convert.py in the pycoal/examples directory
+            4. run example_splib07.py in the pycoal/examples directory
+            5. s07av95a_envi.hdr and s07av95a_envi.sli will be generated
+            
+            Args:
+            library_filename (str): path to Spectral File you wish to convert
+            """
+        if not library_filename:
+            raise ValueError("Must provide path for Spectral File.")
+        
+        with open(library_filename,'r') as f:
+            for i, l in enumerate(f):
+                pass
+            
+        f = open(library_filename,'r')
+        nameTemp = f.readline()
+        name = nameTemp[23:]
+        k = 0
+        x = open('2.txt','w')
+        while(k < i):
+            num = float(f.readline()) * 100
+            num = num / 1000
+            num2 = num * 10
+            num = round(num,4)
+            num2 = round(num2,4)
+            line = str(num) + '  ' + str(num2)
+            x.write(line)
+            x.write('\n')
+            k = k+1
+        f = open(library_filename,'w')
+        f.write('Name:')
+        f.write(name)
+        f.write('Type:\n')
+        f.write('Class:\n')
+        f.write('Subclass:\n')
+        f.write('Particle Size:  Solid\n')
+        f.write('Sample No.:  0095UUUASP\n')
+        f.write('Owner:\n')
+        f.write('Wavelength Range:  ALL\n')
+        f.write('Origin: Spectra obtained from the Noncoventional Exploitation Factors\n')
```

Comment:
This is useless and only works for one specific file. This needs to change entirely.